.github/workflows/dependabot-auto-merge.yml

```diff
@@ -50,8 +50,32 @@ jobs:
           echo "Previous version: ${{ steps.metadata.outputs.previous-version }}"
           echo "New version: ${{ steps.metadata.outputs.new-version }}"
           
-          # Enable auto-merge with merge commit strategy
-          gh pr merge --auto --merge "${{ github.event.pull_request.number }}"
+          # Enable auto-merge using GitHub API (token is automatically masked in logs)
+          response=$(curl -s -w "%{http_code}" -o /tmp/response.json \
+            -X PUT \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GH_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
```

Comment:
@MitaliBhalla If possible, can we avoid using `X-GitHub-Api-Version`?  Looks like if we don't mention it will by default use the same version
https://docs.github.com/en/rest/about-the-rest-api/api-versions?apiVersion=2022-11-28#specifying-an-api-version